src/main/java/core/basesyntax/model/Figure.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.model;
+
+import core.basesyntax.supplier.ColorSupplier;
+
+public abstract class Figure implements Area, Draw {
+    private String color;
+
+    public Figure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        this.color = colorSupplier.getRandomColor();
```

Comment:
you shouldn't do this in the constructor, you should create color for the figure in FigureSupplier and pass it to the constructor of figure heirs